modules/internal/ChapelError.chpl

```diff
@@ -221,23 +247,20 @@ module ChapelError {
            }
 
      */
-    iter these() {
-      var e = _head;
-      while e != nil {
-        yield _to_unmanaged(e);
-        e = e._next;
-      }
+    iter these() ref : owned Error {
+      for i in 0..#nErrors do
+        yield errorsArray[i];
     }
 
-    pragma "no doc"
-    proc deinit() {
-      var e = _head;
-      var todelete: unmanaged Error;
-      while e != nil {
-        todelete = e;
-        e = e._next;
-        delete todelete;
+    /* Returns the first non-nil error contained in this TaskErrors group */
+    proc first() ref : owned Error {
+      var first = 0;
+      for i in 0..#nErrors {
+	if errorsArray[i] != nil {
+          first = i;
+        }
```

Comment:
There's no break in this loop, doesn't that mean this will return the last non-nil error?

Reply:
yes, thanks